### PR TITLE
docs(zh-cn/html): update validation docs

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

update HTML/constraint_validation with fully translated content

### Motivation

Get rid of non-localized content

### CODE REVIEW

File: files/zh-cn/web/html/constraint_validation/index.md [hunk 1]
```diff
@@ -6,393 +6,416 @@ original_slug: Web/Guide/HTML/Constraint_validation
...
+创建 web 表单始终是一个复杂的任务。仅仅组装表单是容易的，但是检查每一个字段的值是否有效并且一致是一件更加困难的事情，而如何告知用户错误所在可能会令人头痛。[HTML5](/zh-CN/docs/Glossary/HTML5) 引入了表单相关的一些新机制：它为 {{ HTMLElement("input") }} 元素和*约束验证*增加了一些新的语义类型，使得客户端检查表单内容变得容易。基本上，通过设置一些新的属性，常用的约束条件可以无需 JavaScript 代码而检测到；对于更复杂的约束条件的校验可以尝试使用约束验证 API。
 
-> **备注：** HTML5 Constraint validation doesn't remove the need for validation on the _server side_. Even though far fewer invalid form requests are to be expected, invalid ones can still be sent by non-compliant browsers (for instance, browsers without HTML5 and without JavaScript) or by bad guys trying to trick your web application. Therefore, like with HTML4, you need to also validate input constraints on the server side, in a way that is consistent with what is done on the client side.
+对于这些概念的基本介绍和示例，参阅[表单验证基础](/zh-CN/docs/Learn/Forms/Form_validation)一页的介绍。
```
Comment (yin1999) [inline files/zh-cn/web/html/constraint_validation/index.md:11]:
```suggestion
对于这些概念的基本介绍和示例，参阅[表单数据校验教程](/zh-CN/docs/Learn/Forms/Form_validation)一页的介绍。
```

File: files/zh-cn/web/html/constraint_validation/index.md [hunk 2]
```diff
@@ -6,393 +6,416 @@ original_slug: Web/Guide/HTML/Constraint_validation
...
 
-- 给 {{ HTMLElement("input") }} 元素的 {{ htmlattrxref("type", "input") }} 特性选择最合适的语义化的值，比如，选择 email 类型将会自动创建一个约束用于检查输入的值是否是一个有效的 e-mail 地址。
-- 设置验证相关的特性值，允许用一种简单的方式来描述基本的约束，而不必要使用 JavaScript。
+- 给 {{ HTMLElement("input") }} 元素的 {{ htmlattrxref("type", "input") }} 属性选择最合适的语义化的值，比如，选择 `email` 类型将会自动创建一个约束，用于检查输入的值是否是一个有效的电子邮件地址。
+- 设置验证相关的属性值，允许用一种简单的方式来描述基本的约束，而不必要使用 JavaScript。
 
 ### 语义的 input 类型
 
-{{ htmlattrxref("type", "input") }} 特性中固有约束：
+{{ htmlattrxref("type", "input") }} 特性中的固有约束有：
```
Comment (yin1999) [inline files/zh-cn/web/html/constraint_validation/index.md:24]: 上下文保持一致

```suggestion
{{ htmlattrxref("type", "input") }} 属性中的固有约束有：
```

File: files/zh-cn/web/html/constraint_validation/index.md [hunk 3]
```diff
@@ -6,393 +6,416 @@ original_slug: Web/Guide/HTML/Constraint_validation
...
-- By a call to the checkValidity() function on the [`HTMLFormElement`](/zh-CN/docs/Web/API/HTMLFormElement) interface, which is called _statically validating the constraints_.
-- By submitting the form itself, which is called _interactively validating the constraints_.
+- 通过调用表单相关的 DOM 接口（[`HTMLInputElement`](/zh-CN/docs/Web/API/HTMLInputElement)、[`HTMLSelectElement`](/zh-CN/docs/Web/API/HTMLSelectElement)、[`HTMLButtonElement`](/zh-CN/docs/Web/API/HTMLButtonElement) 、[`HTMLOutputElement`](/zh-CN/docs/Web/API/HTMLOutputElement) 或 [`HTMLTextAreaElement`](/zh-CN/docs/Web/API/HTMLTextAreaElement)）的 `checkValidity()` 或 `reportValidity()` 方法，只对这个元素进行约束评估，允许脚本获得这些信息。`checkValidity()` 方法返回一个布尔值，表示该元素的值是否通过其约束（这通常是由用户代理在确定哪个 CSS 伪类，{{ Cssxref(":valid") }} 或 {{ Cssxref(":invalid") }} 适用时完成的）。相反，`reportValidity()` 方法会向用户报告任何约束失败的情况。
+- 通过调用 [`HTMLFormElement`](/zh-CN/docs/Web/API/HTMLFormElement) 接口上的 `checkValidity()` 或 `reportValidity()` 方法。
+- 通过提交表单本身。
```
Comment (yin1999) [inline files/zh-cn/web/html/constraint_validation/index.md:252]: 这里少了段话，有时间的话可以补上

**Review by yin1999 — APPROVED** [review]

File: files/zh-cn/web/html/constraint_validation/index.md (hunk 3 above)
Comment (jasonren0403) [inline files/zh-cn/web/html/constraint_validation/index.md:252]:
```suggestion

调用 `checkValidity()` 也被称为约束的*静态*验证，调用 `reportValidity()` 也被称为约束的*交互*认证。
```

**Pull request merged**